src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -532,13 +547,20 @@
      * @param currentRow: custom DataImportRow class data passed from the Apex controller.
      * @return Object row
      */
-    flattenDataImportRow: function(currentRow) {
+    flattenDataImportRow: function(namespace, currentRow) {
         let row = currentRow.record;
         row.donorName = currentRow.donorName;
         row.donorLink = currentRow.donorLink;
         row.matchedRecordUrl = currentRow.matchedRecordUrl;
         row.matchedRecordLabel = currentRow.matchedRecordLabel;
         row.errors = currentRow.errors;
+        // reformat lookup info to display as a url in datatable
```

Comment:
I wonder if it might be worth refactoring the DataImportRow class in apex to support the translation for reference Id to the link+label needed to build the data table link below, especially as we move towards support any lookup fields generically. 

Even here, the operation for the Campaign Link seems similar in theory to the donor link or the matched record link, but the logic for the first two is handled in the apex DataImportRow class, and the campaign link logic is here in javascript. Is there a reason not to do all the three in the same place, either in the DataImportRow class or in this method?

Reply:
Follow-up thought, if we include a list of relationship field info from the Apex side of things, would that be a good setup to move towards handling generic lookup fields, as opposed to handling each specific case differently?

Reply:
The logic for the matching URL/label is a good bit more complex. I'm for any future lookup fields being handled in the JS, but would push back a bit on the system-provided matching field and the lookup fields added by the user being semantically similar. This might also be something to refactor in group review.

Reply:
I think if this is part of a strategic switch to handling in javascript, it's probably ok to use this as a bridge opportunity. Just as long as we are being thoughtful/intentional about it!

Reply:
I wrote this section with an eye toward being able to parse any number of user-defined lookup fields in a loop and construct them here. I'll back Randi up on the fact that the donor and donation ones have additional logic behind them (Account or Contact? Opportunity or Payment?) that likely make them better left back in apex. I'd love to review this as a group and see if there's a better solution!